notebooks/228-clip-zero-shot-image-classification/228-clip-zero-shot-image-classification.ipynb

```diff
@@ -0,0 +1,281 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Zero-shot Image Classification with OpenAI Clip and OpenVINO™\n",
+    "\n",
+    "Zero-shot image classification is a computer vision task to classify images into one of several classes, without any prior training or knowledge of the classes. \n",
+    "\n",
+    "![zero-shot-pipeline](https://user-images.githubusercontent.com/29454499/207773481-d77cacf8-6cdc-4765-a31b-a1669476d620.png)\n",
+    "\n",
+    "Zero-shot learning resolves several challenges in image retrieval systems. For example, with the rapid growth of categories on the web, it is challenging to index images based on unseen categories. With zero-shot learning we can associate unseen categories to images by exploiting attributes to model the relationships among visual features and labels.\n",
+    "\n",
+    "In this tutorial we will use [OpenAI CLIP](https://github.com/openai/CLIP) model for performing zero-shot image classification. Notebook contains following steps:\n",
+    "1. Download model\n",
+    "2. Instantiate PyTorch model\n",
+    "3. Export ONNX model and convert to OpenVINO IR using the Model Optimizer tool\n",
+    "4. Run CLIP with OpenVINO"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Instantiate model\n",
+    "\n",
+    "CLIP (Contrastive Language-Image Pre-Training) is a neural network trained on a variety of (image, text) pairs. It can be instructed in natural language to predict the most relevant text snippet, given an image, without directly optimizing for the task.\n",
+    "CLIP uses a ViT like transformer to get visual features and a causal language model to get the text features. Both the text and visual features are then projected to a latent space with identical dimension. The dot product between the projected image and text features is then used as a similar score.\n",
+    "\n",
+    "![clip](https://raw.githubusercontent.com/openai/CLIP/main/CLIP.png)\n",
```

Comment:
Could you please add a link and/or reference for this image?

Reply:
do you have any examples or guidelines how it should looks like? I want to make it in consistent with other notebooks way

Reply:
added